src/main.rs

```diff
@@ -277,3 +483,34 @@ fn detect_collisions(mut events: EventReader<CollisionEvent>, mut health_query:
         }   
     }  
 }
+
+fn detect_slime_collisions(mut events: EventReader<CollisionEvent>, mut enemy_health_query: Query<&mut Health, (With<Slime>, Without<FlameSpirit>)>) {
+
+    for event in events.iter().filter(|e| e.is_started()) {    
+        let mut health = if let Some(health) = enemy_health_query.iter_mut().next() { health } else { return; };
+        let (layers_1, layers_2) = event.collision_layers();
+        if is_projectile(layers_1) && is_enemy(layers_2) {
+            health.value -= 1;
+            println!("Enemy Health {}", health.value);
+        } else if is_projectile(layers_2) && is_enemy(layers_1) {
+            health.value -= 1;
+            println!("Enemy Health {}", health.value);
+        }   
+    }  
+}
+
+fn detect_flame_collisions(mut events: EventReader<CollisionEvent>, mut enemy_health_query: Query<&mut Health, (With<FlameSpirit>, Without<Slime>)>) {
+
+    for event in events.iter().filter(|e| e.is_started()) {    
+        let mut health = if let Some(health) = enemy_health_query.iter_mut().next() { health } else { return; };
+        let (layers_1, layers_2) = event.collision_layers();
+        if is_projectile(layers_1) && is_enemy(layers_2) {
+            health.value -= 1;
+            println!("Enemy Health {}", health.value);
+        } else if is_projectile(layers_2) && is_enemy(layers_1) {
+            health.value -= 1;
+            println!("Enemy Health {}", health.value);
+        }   
+    }  
+}
+
```

Comment:
This is kind of the same. Why not create a shared tag between all enemies and make the collision based on the shared tag and have this code only once